enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,192 @@
+This enhancement replaces that approach with a new manifest annotation requesting the CVO delete the in-cluster object instead of creating/updating it.
+This will provide a more straightforward way for engineering to remove content.
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
+
+### Goals
+
+* Developers will be able to remove any of the currently managed CVO objects by modifying an existing manifest and adding the new delete annotation.
+
+### Non-Goals
+
+* This enhancement does not add any new objects for management by CVO, therefore the new delete annotation only applies to the currently supported manifests and any that may be added in the future.
+This enhancement neither adds nor requires any application specific delete logic within CVO.
+
+## Proposal
+
+When the following annotation appears in a CVO supported manifest and is set to "true" the associated object will be removed from the cluster by the CVO.
+Values other than `true` will result in CVO logging an error and no further processing of the manifest.
+```
+apiVersion: apps/v1
+...
+metadata:
+...
+  annotations:
+    release.openshift.io/delete: "true"
+```
+The existing CVO ordering scheme defined [here](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/operators.md#what-is-the-order-that-resources-get-createdupdated-in) will also be used for object removal.
+This both minimizes CVO code changes and provides a simple method of deleting multiple objects by reversing the order in which they were created.
+It is the developer's responsibility to ensure proper deletion ordering and to ensure that all items originally created by an object are deleted when that object is deleted.
+For example, an operator may have to be modified to take explicit actions upon delete request to remove external resources it had originally created.
+
+### Delete Finalization
+
+Two delete finalization options are described below to help reach a consensus on the best approach.
+
+#### Finalization Option 1
+
+The first approach is to handle deletion requests similar to how CVO handles create/update requests.
+This is in a non-blocking manner whereby CVO issues the initial request to delete an object kicking off resource finalization and after which resource removal.
+CVO does not wait for actual resource removal but instead continues.
+As CVO manifest processing continues the object's `.metadata.deletionTimestamp`, set when the delete was originally requested, informs CVO that this object has already been processed for deletion.
+If an object cannot be successfully removed cluster update will continue since the object's presence, or partial removal, should not affect cluster operation.
+
+The advantage of this approach is that it does not slow down the processing of the manifest graph. Also, being similar to current CVO manifest graph processing will most likely make it easier to implement and maintain.
```

Comment:
I'm ok with "slow" processing.  My concern, if we block on finalization during graph traversal, is that during updates the CVO will no longer be managing any resources associated with manifests later in the manifest graph.  That could leave large portions of the graph unmanaged for long durations.

Reply:
I updated wording in commit [c65586e](https://github.com/openshift/enhancements/pull/419/commits/c65586e56346ba721dc73c498b3ae1a20c40adc0).